subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/artifactreuse/ArtifactResolutionQueryIntegrationTest.groovy

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.artifactreuse
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import spock.lang.Issue
+import spock.lang.Timeout
+
+import java.util.concurrent.TimeUnit
+
+class ArtifactResolutionQueryIntegrationTest extends AbstractHttpDependencyResolutionTest {
+    BlockingHttpServer server = new BlockingHttpServer()
```

Comment:
If you declare this as a `@Rule`, you don't need to handle start and stop yourself.

Reply:
Yeah, fixed. But I still have to handle `start` myself because `BlockingHttpServer` doesn't override [`before()`](https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/rules/ExternalResource.java#L75). 

Then I tried to override it, but got a another error. I checked other usage of `BlockingHttpServer` and they all choose to handle `start()` themselves so I guess this is intentional.